packages/core/src/StarknetWindowObject.ts

```diff
@@ -0,0 +1,123 @@
+import type { AccountInterface, ProviderInterface } from "starknet"
+
+export type AccountChangeEventHandler = (accounts: string[]) => void
+
+export type NetworkChangeEventHandler = (network?: string) => void
+
+export type WalletEventHandlers =
+  | AccountChangeEventHandler
+  | NetworkChangeEventHandler
+
+export type WalletEvents =
+  | {
+      type: "accountsChanged"
+      handler: AccountChangeEventHandler
+    }
+  | {
+      type: "networkChanged"
+      handler: NetworkChangeEventHandler
+    }
+
+// EIP-747:
+// https://github.com/ethereum/EIPs/blob/master/EIPS/eip-747.md
+export interface WatchAssetParameters {
+  type: "ERC20" // The asset's interface, e.g. 'ERC20'
+  options: {
+    address: string // The hexadecimal StarkNet address of the token contract
+    symbol?: string // A ticker symbol or shorthand, up to 5 alphanumerical characters
+    decimals?: number // The number of asset decimals
+    image?: string // A string url of the token logo
+    name?: string // The name of the token - not in spec
+  }
+}
+
+// EIP-3085
+// https://github.com/ethereum/EIPs/blob/master/EIPS/eip-3085.md
+
+export interface AddStarknetChainParameters {
+  id: string
+  chainId: string // A 0x-prefixed hexadecimal string
+  chainName: string
+  baseUrl: string
+  rpcUrl?: string
+  blockExplorerUrl?: string
```

Comment:
let's make it an array for future proofing (it's `blockExplorerUrls?: string[];` on L1)